src/app/profile/[username]/components/UserProfile.tsx

```diff
@@ -58,30 +64,55 @@ export default function UserProfile({
   useEffect(() => {
     let isCancelled = false;
 
-    const fetchWalletData = async () => {
+    const fetchAndSetWalletData = async () => {
       try {
-        const walletData = await getUserWalletData(username);
-        if (walletData && !isCancelled) {
-          setEthAddress(
-            walletData.wallets.find((wallet) => wallet.chain === "ethereum")
-              ?.address,
-          );
-          setSolAddress(
-            walletData.wallets.find((wallet) => wallet.chain === "solana")
-              ?.address,
-          );
+        const readmeUrl = `https://api.github.com/repos/${username}/${username}/contents/README.md`;
+        const response = await fetch(readmeUrl, {
+          headers: {
+            Accept: "application/vnd.github.v3+json",
+            "User-Agent": "Eliza-Leaderboard-App",
+          },
+        });
+
+        if (!response.ok) {
+          return;
+        }
+
+        const readmeData = await response.json();
+        if (!readmeData.content) {
+          return;
         }
+
+        const decodedReadmeText = decodeBase64(readmeData.content);
+        const walletData = parseWalletLinkingDataFromReadme(decodedReadmeText);
+
+        if (isCancelled || !walletData) {
+          setEthAddress(undefined);
+          setSolAddress(undefined);
+          return;
+        }
+
+        setEthAddress(
+          walletData.wallets.find((wallet) => wallet.chain === "ethereum")
+            ?.address,
+        );
+        setSolAddress(
+          walletData.wallets.find((wallet) => wallet.chain === "solana")
+            ?.address,
+        );
```

Comment:
We shouldnt be doing API calls or network fetches in client components like this. We want it to only generate static prerendered pages based on whats in the DB already.

Also, we should be using the github client classes for any calls to github.

Also, this type of data fetching and setting with useState is obselete, should use tanstack query instead (if its needed, but in this case it shouldnt even be doing anything but rendering from the data in the DB now that we have a walletAddresses table